File: TCP  IP Server Client Example Part 5 (Chat Application)/[C#]-TCP  IP Server Client Example Part 5 (Chat Application)/C#/ChatApp1.2/Forms/FrmRegister.cs
using System;
using System.Linq;
using System.Windows.Forms;

namespace ChatApp1._2.Forms
{
    public partial class FrmRegister : Form
    {
        public string RegisteredUserEmail { get; set; }
        public string RegisteredUserPassword { get; set; }

        public FrmRegister()
        {
            InitializeComponent();
        }

        private void FrmRegister_Load(object sender, EventArgs e)
        {

        }

        // Validate Input
        private void BtnLoginClick(object sender, EventArgs e)
        {
            if (tbEmailAddress.Text.Length > 0)
            {
                if (tbPassword.Text.Length > 0 && tbPassword.Text == tbConfirmPassword.Text)
                {
                    RegisteredUserEmail = tbEmailAddress.Text;
                    RegisteredUserPassword = tbPassword.Text.Replace(",", "");
                    this.Close();
                }
                else
                {
                    MessageBox.Show("Passwords are invalid or do not match");
                }
            }
            else
            {
                MessageBox.Show("Email Address is not Valid");
            }
        }

        private void BtnCancelClick(object sender, EventArgs e)
        {
            this.Close();
        }
    }
}

File: TCP  IP Server Client Example Part 5 (Chat Application)/[VB]-TCP  IP Server Client Example Part 5 (Chat Application)/VB/ChatApp1.2/Forms/FrmRegister.vb
Imports System
Imports System.Linq
Imports System.Windows.Forms

Namespace ChatApp1._2.Forms
	Partial Public Class FrmRegister
		Inherits Form

		Public Property RegisteredUserEmail() As String
		Public Property RegisteredUserPassword() As String

		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub FrmRegister_Load(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.Load

		End Sub

		' Validate Input
		Private Sub BtnLoginClick(ByVal sender As Object, ByVal e As EventArgs) Handles btnLogin.Click
			If tbEmailAddress.Text.Length > 0 Then
				If tbPassword.Text.Length > 0 AndAlso tbPassword.Text = tbConfirmPassword.Text Then
					RegisteredUserEmail = tbEmailAddress.Text
					RegisteredUserPassword = tbPassword.Text.Replace(",", "")
					Me.Close()
				Else
					MessageBox.Show("Passwords are invalid or do not match")
				End If
			Else
				MessageBox.Show("Email Address is not Valid")
			End If
		End Sub

		Private Sub BtnCancelClick(ByVal sender As Object, ByVal e As EventArgs) Handles btnCancel.Click
			Me.Close()
		End Sub
	End Class
End Namespace
